R/derive_merged.R

```diff
@@ -412,7 +430,7 @@ derive_vars_merged <- function(dataset,
       )
     )
   }
-  dataset <- left_join(dataset, add_data, by = vars2chr(by_vars))
+  dataset <- left_join(dataset, add_data, by = vars2chr(by_vars), relationship = relationship)
```

Comment:
The error caused by `relationship` should to be caught to provide a useful message. At the moment we get
```
> derive_vars_merged(
+   vs,
+   dataset_add = select(dm, -DOMAIN),
+   by_vars = exprs(STUDYID, USUBJID),
+   relationship = "one-to-one"
+ ) %>%
+   select(STUDYID, USUBJID, VSTESTCD, VISIT, VSSTRESN, AGE, AGEU)
Error in `dplyr::left_join()`:
! Each row in `y` must match at most 1 row in `x`.
ℹ Row 1 of `y` matches multiple rows in `x`.
Run `rlang::last_trace()` to see where the error occurred.
```
For the user it is not clear to what `x` and `y` are referring to.

Reply:
@bundfussr @ddsjoberg Is anyone able to provide some advice on how I 'catch' the error in order to write a better one?

Reply:
Thank you very much - will take a look and implement that now!

Reply:
Implemented.